Libraries/SDHC/ff13/Source/mscmem.c

```diff
@@ -0,0 +1,339 @@
+/******************************************************************************
+ * Copyright (C) 2023 Maxim Integrated Products, Inc., All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL MAXIM INTEGRATED BE LIABLE FOR ANY CLAIM, DAMAGES
+ * OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * Except as contained in this notice, the name of Maxim Integrated
+ * Products, Inc. shall not be used except as stated in the Maxim Integrated
+ * Products, Inc. Branding Policy.
+ *
+ * The mere transfer of this software does not imply any licenses
+ * of trade secrets, proprietary technology, copyrights, patents,
+ * trademarks, maskwork rights, or any other form of intellectual
+ * property whatsoever. Maxim Integrated Products, Inc. retains all
+ * ownership rights.
+ *
+ ******************************************************************************/
+
+/**
+ * @file    mscmem.h
+ * @brief   Memory routines used by the USB Mass Storage Class example.
```

Comment:
@sanjayjaroli can you take a look at the application of these files towards the SDHC library?  Are there any maintenance considerations?

@MertCanBoz1 since FatFS is not USB-specific I wonder if using the [Libraries/MiscDrivers/ExtMemory](https://github.com/Analog-Devices-MSDK/msdk/tree/main/Libraries/MiscDrivers/ExtMemory) is a better choice?

Reply:
It may be, but the external flash on the connectivity board of the MAXREFDES178 is different than the one used for EvKits. Some functions (specifically Ext_Flash_Erase.c ) need different handling than the ones at the mx25.c. The new AppSwitcher code calls mx25.c directly from the source file ( https://github.com/MaximIntegratedAI/refdes/pull/48 ).
Therefore, adding the mscmem.c to the [Libraries/MiscDrivers/ExtMemory](https://github.com/Analog-Devices-MSDK/msdk/tree/main/Libraries/MiscDrivers/ExtMemory) may cause some conflicts between them.

Reply:
I think that's what I mean - we should use mx25.c directly.  Ext_Flash.h provides a struct to abstract the physical communication layer which should be enough to address connectivity issues I think.  

https://github.com/Analog-Devices-MSDK/msdk/blob/9bf68cf780920cc3d9654418c5722543f2e69c89/Libraries/MiscDrivers/ExtMemory/Ext_Flash.h#L143

Maybe I missed something - Does the cube camera present itself as a mass storage device for USB?  Is that what the mscmem files are used for?